src/funcx_common/messagepack/README.md

```diff
@@ -0,0 +1,111 @@
+# messagepack
+
+This subpackage defines a message passing protocol for funcx endpoints and the
+forwarder to exchange information, called "messagepack".
+
+Today, only the v0 implementation of the protocol exists, but under wrappers
+which, in theory, could be used to support future protocol versions.
+
+## Protocol Support, send and receive
+
+Message sending specifies a protocol version defaulting to the lowest supported
+version of the protocol in use. This ensures a broad range of compatibility
+with readers of the messages.
+
+When receiving a message, the first step is always to attempt to determine the
+protocol version.
+
+## Version Detection (v0 vs other versions)
+
+In v0 of the protocol, messages are packed bytes with no header, wrapper, or
+other version information.
+
+Under future versions, an explicit protocol version field will be added.
+
+As a result, the detection logic, until the removal of v0, will be as follows:
+- Attempt to detect a protocol_version
+- If detection fails, protocol_version=0
+- Otherwise, protocol_version=<detected>
+
+## Proposed Protocol Version 1
```

Comment:
I had a brief conversation with Yadu about what a next version might look like. We might actually want to use something like FlatBuffers. But I think JSON or JSON with a short (e.g. 4-byte) header would be a reasonable next step. I'm leaving this here as it still is a valuable example of the sort of thing that we might do next.